src/macos.rs

```diff
@@ -19,28 +19,38 @@ use core_foundation::string::CFString;
 use objc::{msg_send, sel, sel_impl};
 
 use crate::{GlConfig, GlError, Profile};
+use cocoa::foundation::NSAutoreleasePool;
+use objc::runtime::Object;
 
 pub struct GlContext {
-    view: id,
     context: id,
 }
 
 impl GlContext {
     pub fn create(
-        parent: &impl HasRawWindowHandle,
+        parent: &dyn HasRawWindowHandle,
         config: GlConfig,
+        shared_context: Option<&GlContext>,
     ) -> Result<GlContext, GlError> {
         let handle = if let RawWindowHandle::MacOS(handle) = parent.raw_window_handle() {
             handle
         } else {
             return Err(GlError::InvalidWindowHandle);
         };
 
-        if handle.ns_view.is_null() {
+        let ns_view = if !handle.ns_view.is_null() {
```

Comment:
Some providers of HasRawWindowHandle do not populate ns_view. This logic is the same as raw-window-metal  https://github.com/norse-rs/raw-window-metal/blob/b85108fc474834c0625a54f3cadd7fcea1103a8f/src/macos.rs#L8.